docs/blog/12factor.md

```diff
@@ -0,0 +1,219 @@
+
+This factor is nevertheless, the key to the always-on experience that we take for granted in large cloud services.
+
+Indicators:
+
+- long delay or elevated error rate when deploying because load balancers misjudge the app's readiness
+- manual steps in deployment even when deployments are nominally automated
+- deploys only during off-hours
+
+## Factor X: Dev/prod parity
+
+> The twelve-factor app is designed for continuous deployment by keeping the gap between development and production small.
+
+This factor has still not established itself fully in that many production environments are hard to squeeze onto a developer's laptop. Generally speaking, the public cloud providers put too little effort into supporting development use cases for their services.
+
+[Kind](https://kind.sigs.k8s.io/) deserves mentioning for its heroic effort to achieve a dev-friendly, multi-node Kuberentes cluster using only Docker.
+
+Docker has introduced a borderland where it is possible to develop a container using just Docker Engine for dev environment and still be reasonably confident that it will execute properly in e.g. Kubernetes. Still, some provisioning of backend services is still needed and time is wasted maintaining two different sets of instrumentation. For example, apps often have a Docker Compose file to get developers started, and a Kubernetes manifest for test/prod.
+
+> The twelve-factor developer resists the urge to use different backing services between development and production
+
+Many full-stack development setup includes "dev" servers whose role is to automatically reload the app as its source code change. Similarly, tools like Docker Desktop and Tilt provide capabilities and environments that are subtly different from e.g. Azure Container instances or Kubernetes. All these will color developers' choices and risk introducing issues that will not be discovered until late in the process.
+
+Indicators:
+
+- "it works on my machine"
+- docker-compose files in app repositories
+
+## Factor XI: Logs
+
+> A twelve-factor app never concerns itself with routing or storage of its output stream
+
+Interestingly, this factor does not actually advise on the use of logging. Rather it treats them much as pre-contraceptive times viewed children: as something that inevitably accrues as a result of marriage.
```

Comment:
This may be my lacking vocabulary, but is `accrues` the word to use here? I would have used `occurs` but I have to be honest that I had to look up the word accrues

Reply:
I'll change to accumulate to be less obscure.